sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationsSuite.scala

```diff
@@ -384,239 +384,166 @@ class UnsupportedOperationsSuite extends SparkFunSuite {
     outputMode = Append
   )
 
-  // Inner joins: Multiple stream-stream joins supported only in append mode
-  testBinaryOperationInStreamingPlan(
-    "single inner join in append mode",
-    _.join(_, joinType = Inner),
-    outputMode = Append,
-    streamStreamSupported = true)
-
-  testBinaryOperationInStreamingPlan(
-    "multiple inner joins in append mode",
-    (x: LogicalPlan, y: LogicalPlan) => {
-      x.join(y, joinType = Inner).join(streamRelation, joinType = Inner)
-    },
-    outputMode = Append,
-    streamStreamSupported = true)
-
-  testBinaryOperationInStreamingPlan(
-    "inner join in update mode",
-    _.join(_, joinType = Inner),
-    outputMode = Update,
-    streamStreamSupported = false,
-    expectedMsg = "is not supported in Update output mode")
-
-  // Full outer joins: only batch-batch is allowed
-  testBinaryOperationInStreamingPlan(
-    "full outer join",
-    _.join(_, joinType = FullOuter),
-    streamStreamSupported = false,
-    batchStreamSupported = false,
-    streamBatchSupported = false)
+  // stream-stream join
+  {
+    // Inner joins: Multiple stream-stream joins supported only in append mode
+    testBinaryOperationInStreamingPlan(
+      "single inner join in append mode",
+      _.join(_, joinType = Inner),
+      outputMode = Append)
+
+    testBinaryOperationInStreamingPlan(
+      "multiple inner joins in append mode",
+      (x: LogicalPlan, y: LogicalPlan) => {
+        x.join(y, joinType = Inner).join(streamRelation, joinType = Inner)
+      },
+      outputMode = Append)
+
+    testBinaryOperationInStreamingPlan(
+      "inner join in update mode",
+      _.join(_, joinType = Inner),
+      outputMode = Update,
+      streamStreamSupported = false,
+      expectedMsg = "is not supported in Update output mode")
+
+    // Full outer joins: only batch-batch is allowed
+    testBinaryOperationInStreamingPlan(
+      "full outer join",
+      _.join(_, joinType = FullOuter),
+      streamStreamSupported = false,
+      batchStreamSupported = false,
+      streamBatchSupported = false)
+
+    // Left outer, left semi, left anti join: *-stream not allowed
+    Seq((LeftOuter, "LeftOuter join"), (LeftSemi, "LeftSemi join"), (LeftAnti, "Left anti join"))
+      .foreach { case (joinType, name) =>
+        testBinaryOperationInStreamingPlan(
+          name,
+          _.join(_, joinType = joinType),
+          batchStreamSupported = false,
+          streamStreamSupported = false,
+          expectedMsg = name)
+      }
 
-  // Left outer joins: *-stream not allowed
-  testBinaryOperationInStreamingPlan(
-    "left outer join",
-    _.join(_, joinType = LeftOuter),
-    batchStreamSupported = false,
-    streamStreamSupported = false,
-    expectedMsg = "outer join")
+    // Right outer joins: stream-* not allowed
+    testBinaryOperationInStreamingPlan(
```

Comment:
Note: this test is from https://github.com/apache/spark/blob/master/sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationsSuite.scala#L566 .